### PR TITLE
Add support for dynamic parameters

### PR DESCRIPTION
**Pull request by theghost5800**

### CODE REVIEW

File: multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/DynamicParameterUtil.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    public static final String PATTERN_FOR_DYNAMIC_PARAMETERS = "'{'ds/{0}/{1}'}'";
+    public static final String REGEX_PATTERN_FOR_DYNAMIC_PARAMETERS = "^\\{ds[A-Za-z0-9_\\-\\.\\/]+\\}$";
+
+    private DynamicParameterUtil() {
+    }
+
+    public static String getRelationshipName(String dynamicParameter) {
+        return dynamicParameter.substring(StringUtils.ordinalIndexOf(dynamicParameter, "/", 1) + 1,
+                                          StringUtils.ordinalIndexOf(dynamicParameter, "/", 2));
```
Comment (radito3) [inline multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/DynamicParameterUtil.java:15]: If we know that there will only be 2 forward slashes, why not use 
```suggestion
        return dynamicParameter.substring(dynamicParameter.indexOf('/') + 1, dynamicParameter.lastIndexOf('/'));
```

File: multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/DynamicParameterUtil.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    public static String getParameterName(String dynamicParameter) {
+        return dynamicParameter.substring(StringUtils.ordinalIndexOf(dynamicParameter, "/", 2) + 1, dynamicParameter.lastIndexOf("}"));
```
Comment (radito3) [inline multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/DynamicParameterUtil.java:19]: Same here.
This _`ordinalIndexOf`_ seems overly complicated of a name, when we really just need the first and/or last index of the slash

File: multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/Messages.java [hunk 3]
```diff
@@ -47,7 +47,7 @@ private Messages() {
     public static final String CIRCULAR_RESOURCE_DEPENDENCIES_DETECTED = "Resources \"{0}\" and \"{1}\" both depend on each other for processing";
     public static final String RESOURCE_DOES_NOT_EXIST = "Resource \"{0}\" does not exist. Cannot create dependency of non-existent resource.";
     public static final String SELF_REQUIRED_RESOURCE = "Resource \"{0}\" is required by itself for processing";
-    public static final String ILLEGAL_REFERENCES_DETECTED = "Module \"{0}\" does not contain a required dependency for \"{1}\", but contains references to its properties";
+    public static final String ILLEGAL_REFERENCES_DETECTED = "Module/Resource \"{0}\" does not contain a required dependency for \"{1}\", but contains references to its properties";
```
Comment (ikasarov) [inline multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/Messages.java:50]: Good catch!

File: multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/DynamicParameterUtil.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package org.cloudfoundry.multiapps.mta.util;
+
+import org.apache.commons.lang3.StringUtils;
+
+public final class DynamicParameterUtil {
+
+    public static final String PATTERN_FOR_DYNAMIC_PARAMETERS = "'{'ds/{0}/{1}'}'";
+    public static final String REGEX_PATTERN_FOR_DYNAMIC_PARAMETERS = "^\\{ds[A-Za-z0-9_\\-\\.\\/]+\\}$";
```
Comment (ikasarov) [inline multiapps-mta/src/main/java/org/cloudfoundry/multiapps/mta/util/DynamicParameterUtil.java:8]: Maybe enforce only two forward slashes in the regex

**Pull request merged**